### PR TITLE
Fix rz_path_prefix not working when BINDIR is at the root directory or several folders deep

### PR DESCRIPTION
**Pull request by yossizap**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fix `rz_path_prefix` not working when BINDIR is at the root directory or several folders deep.

This change aims to solve [#2897](https://github.com/rizinorg/cutter/issues/2897) - the Cutter build artifact fails to find the themes directory due to changes in [rz_path_prefix](https://github.com/rizinorg/rizin/blob/dev/librz/util/path.c#L25), which only returns the relative prefix for portable binaries that return true for `rz_str_endswith(<pid path>, RZ_SYS_DIR RZ_BINDIR)`. Since Windows Cutter sets `RZ_BINDIR=.` and Linux Cutter runs from the root directory while `RZ_BINDIR=bin`, the original build directory prefix is returned in all cases.

**Test plan**

- Check that Rizin is able to load themes with `-Dbindir=multiple/prefixes`
- Check that Rizin is able to load themes with `-Dbindir=.`
- Verify that [#2897](https://github.com/rizinorg/cutter/issues/2897) is solved by these changes.

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
```
 "cl" "-Ilibrz\util\rz_util.lib.p" "-I." "-ID:\a\rizin\rizin" "-Ilibrz" "-ID:\a\rizin\rizin\librz" "-Ilibrz\include" "-ID:\a\rizin\rizin\librz\include" "-Isubprojects\sdb" "-ID:\a\rizin\rizin\subprojects\sdb" "-Isubprojects\sdb\src" "-ID:\a\rizin\rizin\subprojects\sdb\src" "-Isubprojects\zlib-1.2.11" "-ID:\a\rizin\rizin\subprojects\zlib-1.2.11" "/MT" "/nologo" "/showIncludes" "/utf-8" "/W2" "/O2" "/Gw" "-DRZ_PLUGIN_INCORE=1" "/Fdlibrz\util\rz_util.lib.p\D__a_rizin_rizin_librz_util_path.c.pdb" /Folibrz/util/rz_util.lib.p/D__a_rizin_rizin_librz_util_path.c.obj "/c" D:/a/rizin/rizin/librz/util/path.c
D:/a/rizin/rizin/librz/util/path.c(225): error C2059: syntax error: 'return'
D:/a/rizin/rizin/librz/util/path.c(226): error C2059: syntax error: '}'
```

File: librz/util/path.c
Comment (XVilka) [inline librz/util/path.c]: Missing opening `{`, also broken indentation.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/util/path.c
Comment (GustavoLCR) [inline librz/util/path.c]: Use `GetFullPathNameW()` and use `rz_utf8_to_utf16()` and `rz_utf16_to_utf8()` to convert from `char *` -> `wchar_t *`->`char *`, currently it will break if `UNICODE` is defined, and its better to always support Unicode paths, whether `UNICODE` is defined or not.

**Review by GustavoLCR — CHANGES_REQUESTED** [review]

File: librz/util/path.c
Comment (wargio) [inline librz/util/path.c]:
```suggestion
	if (len > 0 && len < MAX_PATH - 1) {
```

File: librz/util/path.c
Comment (ret2libc) [inline librz/util/path.c]: Instead of this i would use HAVE_REALPATH computed from meson

Comment (yossizap) [reply librz/util/path.c]: Also went ahead and added it in file.c

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]
wait for ret2libc approval

File: librz/util/path.c
Comment (ret2libc) [inline librz/util/path.c]: since this seems to be used only in one place, isn't it better to just use `RZ_JOIN_2_PATHS(RZ_PREFIX, RZ_BINDIR)` there instead of this define with 2 lines comment?

File: meson.build
Comment (ret2libc) [inline meson.build:222]: bindir_depth should be somehow used as well in :
```
if use_rpath
  rpath_exe = '$ORIGIN/../' + get_option('libdir')
  rpath_lib = '$ORIGIN'
  rpath_summary = 'relative'
elif use_rpath_absolute
```

As you can see it assumes that the binaries are in `<top-prefix>/bin`, but this is not true. That should be fixed as well to ensure proper rpath.

**Review by ret2libc — COMMENTED** [review]
Just these two comments. Anyway, rz_path_prefix logic looks good.

File: meson.build
Comment (yossizap) [reply meson.build:222]: I had to set the minimum meson version to 0.58.0 for range() support

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**
